framework/web/auth/CWebUser.php

```diff
@@ -499,19 +499,21 @@ protected function renewCookie()
 		}
 	}
 
-	/**
-	 * Saves necessary user data into a cookie.
-	 * This method is used when automatic login ({@link allowAutoLogin}) is enabled.
-	 * This method saves user ID, username, other identity states and a validation key to cookie.
-	 * These information are used to do authentication next time when user visits the application.
-	 * @param integer $duration number of seconds that the user can remain in logged-in status. Defaults to 0, meaning login till the user closes the browser.
-	 * @see restoreFromCookie
-	 */
-	protected function saveToCookie($duration)
+    /**
+     * Saves necessary user data into a cookie.
+     * This method is used when automatic login ({@link allowAutoLogin}) is enabled.
+     * This method saves user ID, username, other identity states and a validation key to cookie.
+     * These information are used to do authentication next time when user visits the application.
+     * @param integer $duration number of seconds that the user can remain in logged-in status. Defaults to 0, meaning login till the user closes the browser.
+     * @param string $sameSite
+     * @see restoreFromCookie
+     */
+	protected function saveToCookie($duration, $sameSite = CHttpCookie::SAME_SITE_NONE)
```

Comment:
Hi @maznick ,

We don't need to change yii core in this case. Yii can be extended by a number of ways, one of which is to `extend` CWebUser, which is what we have done with our `WebUser` component that re-implements the `saveToCookie` function. Please make your enhancements in our clone of https://github.com/PeoplePerHour/PPH-Yii/blob/master/components/WebUser.php#L527 instead of Yii core.

The more customisations we make to Yii core, the harder it is to maintain patches when the community does updates.